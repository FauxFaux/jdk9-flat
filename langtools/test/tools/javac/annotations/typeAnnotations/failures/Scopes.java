/*
 * @test /nodynamiccopyright/
 * @bug 6843077 8006775
 * @summary check that A is accessible in the class type parameters
 * @author Mahmood Ali
 * @compile/fail/ref=Scopes.out -XDrawDiagnostics Scopes.java
 */
class Scopes<T extends @UniqueInner Object> {
  // UniqueInner is not visible in the type parameters.
  // One has to use Scopes.UniqueInner.
  // Annotations with the default @Target are not allowed there,
  // so we also get the second error about the invalid location.
  // Adding the target here doesn't matter, as we don't resolve
  // the annotation type.
  // @Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
  @interface UniqueInner { };
}
