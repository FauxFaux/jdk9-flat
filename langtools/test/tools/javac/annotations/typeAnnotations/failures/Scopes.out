Scopes.java:8:25: compiler.err.cant.resolve: kindname.class, UniqueInner, , 
Scopes.java:8:24: compiler.err.annotation.type.not.applicable
2 errors
